src/redux/categories/categoriesSlice.js

```diff
@@ -0,0 +1,22 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  catagories: [],
+};
+
+const categoriesSlice = createSlice({
+  name: 'category',
+  initialState,
+  reducers: {
+    checkStatus: (state) => {
+      if (state.catagories === []) {
+        return 'Under construction!';
+      }
+      return state.catagories;
+    },
+  },
+});
```

Comment:
- [ ] Ouch! this is close but I noticed some issues in the `checkStatus` reducer. In Redux, reducers should update the state based on the action payload received, rather than checking the state value itself. To fix this, consider accepting an action as a parameter in the `checkStatus` reducer and checking the payload of that action to update the state. This will ensure that the state is updated correctly and consistently based on the actions dispatched to the store.
- It should be something like the below 👇
```jsx
  reducers: {
    checkStatus: (state, action) => {
      state.catagories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.catagories
    },
  },

```